### ISSUE TITLE
Support Excel tables in `read_excel`

### ISSUE THREAD

**Issue by wdoppenberg** [pola-rs/polars#18949]
### Description

Now that `fastexcel` supports loading individual tables from Excel files (see https://github.com/ToucanToco/fastexcel/pull/287), it would be nice to support it in the `read_excel` method. I see that `ExcelReader.load_sheet_by_name` is used internally, so a design choice for choosing `ExcelReader.load_table` through `read_options` needs to be made. 

I'd love to help out but I'd like to get some input on the parameter design for `read_excel`.

**Comment by alexander-beedie** [comment]
Yup, I just noticed that was out; will check how it was implemented and see if we can integrate👌

**Comment by PrettyWood** [comment]
FYI v0.12.0 is out (thanks again @wdoppenberg for the contribution!)
It should now be possible to integrate that with polars
